server/messages/messageController.js

```diff
@@ -1,45 +1,71 @@
 'use strict';
 // built-in npm module dependencies
-var Promise = require('bluebird');
+//var Promise = require('bluebird');
+//var Promise = require('mpromise');
+var _ = require('underscore');
 
 // local dependencies
 var Message = require('./messageModel.js');
 
 module.exports = {
   addMessage: function (request, response) {
-  // // logic to add or update Message after server receives POST request
-  //   // convert phone number to number
-  //   var messageObject = {
-  //     userId: request.body.user,
-  //     // may have to convert the contactId to an ObjectId to display the relationship
-  //     // see: mongoosejs.com/docs/populate.html
-  //     contactId: request.body.contactId,
-  //     contactPhone: request.body.phone,
-  //     text: request.body.message,
-  //     date: request.body.date,
-  //     status: 'scheduled'
-  //   };
-  //   var newMessage = new Message(messageObject);
-  //   newMessage.save(function (error) {
-  //     if (error) { console.log(error); }
-  //   });
+  // logic to add or update Message after server receives POST request
+    var messageObject = {
+      userId: request.body.userId,
+      contactId: request.body.contactId,
+      text: request.body.text,
+      date: new Date(),
+      status: 'scheduled'
+    };
+    console.log(messageObject);
+    // see: http://mongoosejs.com/docs/models.html
+    var newMessage = new Message(messageObject);
+    newMessage.save(function (error, doc) {
+      if (error) {
+        response.status(500).send('Error: Could not save message');
+      } else {
+        response.status(200).send(doc);
+      }
+    });
   },
 
   updateMessage: function (request, response) {
-    // var sentMessagesArray = request.body.sent;
-    // for (var i = 0; i < sentMessagesArray.length; i++) {
-    //   var query = { id: sentMessagesArray[i] };
-    //   Message.findOneAndUpdate(query, { status: 'sent' }, function (data) {
-        
-    //   });
-    // }
+    //update message in database after receiving message status
+    for (var status in request.body) {
```

Comment:
@kevhuang The incoming PUT request to api/messages/ should be in this format: 
```javascript
{ sent: [messageId, messageId],  
  deleted: [messageId, messageId]
}
```
Line 34 iterates through all properties in this object (i.e., 'sent' and 'deleted') and Line 35 assigns the array of sent or deleted messages to a variable. 

@jyeg - let me know if this accurately reflects the format of the request, or if there are changes required.